### PR TITLE
[4.0] 29957 css

### PR DESCRIPTION
**Pull request by N6REJ**
Pull Request for Issue #29957

### Summary of Changes

icons now center properly

### Testing Instructions
go to template styles and duplicate the atum template.
verify gold star and gray star are vertically aligned.
NPM CI REQUIRED!

### Actual result BEFORE applying this Pull Request

image

### Expected result AFTER applying this Pull Request

image

### Documentation Changes Required
none

### CODE REVIEW

**Comment by N6REJ** [comment]
@infograf768 do you think it would be better to use auto instead of static size?

**Comment by SharkyKZ** [comment]
Doesn't look like the right solution. The issue here is that `icon-color-featured` class is applied twice:

Untitled

**Comment by hans2103** [comment]
icon is set by:
https://github.com/joomla/joomla-cms/blob/529aeb7c8019009c7e50a3af112a784a1784f01d/administrator/components/com_templates/tmpl/styles/default.php#L98

`jgrid.isdefault` is a function from: 
https://github.com/joomla/joomla-cms/blob/529aeb7c8019009c7e50a3af112a784a1784f01d/libraries/src/HTML/Helpers/JGrid.php#L348

and that one calls `static::state` from the same file:
https://github.com/joomla/joomla-cms/blob/529aeb7c8019009c7e50a3af112a784a1784f01d/libraries/src/HTML/Helpers/JGrid.php#L205

and that one calls `static::action` from the same file:
https://github.com/joomla/joomla-cms/blob/529aeb7c8019009c7e50a3af112a784a1784f01d/libraries/src/HTML/Helpers/JGrid.php#L45

In that last function the magic happens.
the `$active_class` is set to both the anchor as the span inside that anchor.
`$active_class="icon-color-featured fas fa-star"`

I agree with @SharkyKZ that css is not the right solution. Changing the variables in the PHP should do the trick. 
@N6REJ can you do that?

**Comment by N6REJ** [comment]
@hans2103 not sure what should change.  Would you mind making a pr for that?  Or correcting this pr?

**Comment by N6REJ** [comment]
@SharkyKZ the css fix was needed because they used different px sizing.  When elaine did the whole color-featured thing I thought it was weird but she's the design person not me.  
And of course the fa- should match icon- which it didn't.

**Comment by adj9** [comment]
I haven't tried npm, but with only PR application the stars aren't cheered.
With the inspection of HTML you have this code
Schermata 2020-07-11 alle 12 42 50

The first tag <tr> open is from the yellow star and the other is from the grey star.

File: administrator/templates/atum/scss/blocks/_icons.scss [hunk 1]
```diff
@@ -21,7 +21,8 @@
   }
 }
 
-.icon-unpublish {
+.icon-unpublish,
+.fa-times{
```
Comment (Quy) [inline administrator/templates/atum/scss/blocks/_icons.scss]: Add space before `{`. Repeat in other instances in this PR.

**Comment by N6REJ** [comment]
@Quy @hans2103 @SharkyKZ Thanks for the fixes.. Got it all squared away now.

**Comment by Quy** [comment]
Please fix.

**Comment by N6REJ** [comment]
> Please fix.
> 

what in the world!!!!!!

**Comment by hans2103** [comment]
I can reproduce the issue. 

# before applying PR
and running `npm build:css`
Schermafdruk 2020-07-15 19 53 12

# after applying PR
and running `npm build:css`
Schermafdruk 2020-07-15 19 54 43

**Comment by hans2103** [comment]
@N6REJ your suggested changes to fix styling on the default star have some negative effect on the styling of all icons on the website so it seems. :-)
Back to the drawing table I guess

**Comment by N6REJ** [comment]
> @N6REJ your suggested changes to fix styling on the default star have some negative effect on the styling of all icons on the website so it seems. :-)
> Back to the drawing table I guess

thats what I'm looking @.. it's doing it even in 4.0-dev branch on my end but I'm going to test things more carefully

**Comment by N6REJ** [comment]
not my bug 😀 
running composer install then npm ci on fresh pull of 4.0-dev fixed it.

**Comment by N6REJ** [comment]
as you can see all the broken icons were changed recently.. 
image

**Comment by hans2103** [comment]
@N6REJ then please merge most recent changes of branch `4.0 dev` into your pull request and check again.

**Comment by N6REJ** [comment]
@hans2103 exactly what I did, as I explained.  Here you can see that I've done so.

https://youtu.be/j-4Rqk2UQhs
Pushed AFTER this video cause I forgot to during it.

**Comment by hans2103** [comment]
@N6REJ please point me where you run `npm build:css` after applying your changes.
See my video where I run `npm build:css` both before and after applying your changes
https://www.youtube.com/watch?v=BB6NEX06n_k

**Comment by N6REJ** [comment]
@hans2103 I don't.. I ran npm ci !  
Nice trick with storm & npm module on left.  How'd you get a npm module like that?
I'm going to try a pre-built package and see if that works or not.  I'm also going to do my best to find out exactly why PT isn't working.  
I haven't tried it with PT.

**Comment by N6REJ** [comment]
@hans2103 idk whats wrong with my local system but I can confirm that my pr is currently broken using the pre-built.
what I don't understand is why it works fine locally.
I've got storm set to the pr branch 
image
and you can see its fine...
image
Here's my install..
https://hallhome.us/joomla-cms.zip
@richard67 @roland-d any clue whats going on?
back to the drawing board for me... *sigh*

**Comment by hans2103** [comment]
> Nice trick with storm & npm module on left. How'd you get a npm module like that?

PHP Storm > right mouse click on file package.json and select option "Show NPM Scripts"

**Comment by N6REJ** [comment]
@Quy @hans2103 @richard67
I'm happy to report I FINALLY found the cause.  I'm so sorry for having such a crappy pr.
I'll be much more careful in the future.  I couldn't see the error on chrome but it was very clear in FF-Dev.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on cc41d1e58e69e819ab0168f5f2d828602762c976<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29983">issues.joomla.org/tracker/joomla-cms/29983</a>.</sub>

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on cc41d1e58e69e819ab0168f5f2d828602762c976<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29983">issues.joomla.org/tracker/joomla-cms/29983</a>.</sub>

**Comment by infograf768** [comment]
rtc<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29983">issues.joomla.org/tracker/joomla-cms/29983</a>.</sub>

**Comment by infograf768** [comment]
Tks.

**Pull request merged**